samples/durable-task-sdks/dotnet/Agents/PromptChaining/Worker/Services/AgentServices.cs

```diff
@@ -0,0 +1,433 @@
+3. Include key facts from the research in order of importance
+4. Cite sources appropriately within the text using journalistic attribution standards
+5. Organize the article with logical structure following inverted pyramid style
+6. Write in a balanced, objective tone adhering to journalistic standards
+7. Include quotes if available in the research with proper attribution
+8. End with a conclusion that summarizes or provides perspective
+
+The article should be approximately 400-600 words in length.
+Format the article with appropriate HTML tags (<h1>, <p>, etc.) following journalistic standards.";
+
+        Logger.LogInformation($"Requesting article creation for topic: {topic}");
+        return await GetResponseAsync(prompt);
+    }
+}
+
+/// <summary>
+/// Image generation agent service for news articles with direct DALL-E integration
+/// </summary>
+public class ImageGenerationAgentService : BaseAgentService
+{
+    private const string AgentName = "ImageGenerationAgent";
+    private const string EndpointEnvVar = "AGENT_CONNECTION_STRING";
+    private readonly string _systemPrompt = @"You are an expert image specialist for news articles.
+Create detailed descriptions of images that would complement news stories.
+Focus on photorealistic, journalistically appropriate imagery and compositions.
+Provide descriptive captions that enhance understanding of the article content.";
+    
+    private bool _initialized = false;
+    private readonly HttpClient? _httpClient;
+    private readonly string? _dallEEndpoint;
+    private readonly bool _dalleEnabled;
+    private readonly DefaultAzureCredential? _credential;
+    
+    public ImageGenerationAgentService(ILogger<ImageGenerationAgentService> logger) 
+        : base(Environment.GetEnvironmentVariable(EndpointEnvVar) ?? 
+              throw new InvalidOperationException($"Environment variable '{EndpointEnvVar}' not set"), 
+              logger)
+    {
+        // Check if DALL-E endpoint is set
+        string? dalleEndpoint = Environment.GetEnvironmentVariable("DALLE_ENDPOINT");
+        
+        // Only enable DALL-E if endpoint is provided
+        _dalleEnabled = !string.IsNullOrEmpty(dalleEndpoint);
+        
+        if (_dalleEnabled)
+        {
+            Logger.LogInformation($"DALL-E image generation is enabled with Microsoft Entra ID authentication");
+            Logger.LogInformation($"Using DALL-E endpoint: {dalleEndpoint}");
+            
+            // Use the DALL-E endpoint environment variable directly without modification
+            // Note: The DALLE_ENDPOINT should be the complete URL including deployment name and API version
+            // Example format: https://your-resource.openai.azure.com/openai/deployments/your-deployment-name/images/generations?api-version=2023-12-01-preview
+            _dallEEndpoint = dalleEndpoint;
+            
+            // Create HTTP client
+            _httpClient = new HttpClient();
```

Comment:
There is a potential issue of leaking network resources by creating `HttpClient` objects and never disposing them. See [here](https://learn.microsoft.com/aspnet/core/fundamentals/http-requests) for best practices related to this.